File: python/programmers/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    
    arr = []
    for p in moves:
        for row in board:
            if row[p-1] != 0:
                if len(arr) != 0 and row[p-1] == arr[-1]:
                    arr.pop()
                    answer += 2
                else:
                    arr.append(row[p-1])
                
                row[p-1] = 0
                break
    
    return answer

File: src/main/kotlin/programmers/kakao_intern/KI2019/크레인 인형뽑기 게임.kt
package programmers.KI2019.크레인인형뽑기게임

// 26분

import java.util.Stack

class Solution {
    fun solution(board: Array<IntArray>, moves: IntArray): Int {
        var result = 0
        val basket = Stack<Int>()

        moves.forEach { j ->
            for (i in 0 until board.size) {
                val doll = board[i][j - 1]
                if (doll > 0) {
                    board[i][j - 1] = 0
                    if (basket.isNotEmpty() && basket.peek() == doll) {
                        basket.pop()
                        result += 2
                    } else {
                        basket.add(doll)
                    }
                    break
                }
            }
        }

        return result
    }
}